### DISCUSSION TITLE
Issue with default values on arrays

### DISCUSSION THREAD

**Discussion by daniele-baccega** [FLAMEGPU/FLAMEGPU2#1207] [Q&A] [answered]
Hi,
I have a problem to set default values on arrays at definition time. I have to initialize an array of lots of elements, but I don't know how.
I tried this:
pedestrian.newVariable<int, 100000>(FLOW, {-1});

But in this way only the first element is initialized to -1, all the other elements to 0. is there a way to initialize an array in a simple way at definition time?

Thank you

**Comment by Robadob** [comment, marked as answer]
*That's a much larger array that we ever really expected to be used for agent variables.*

The `newVariable()` [expects the default](https://github.com/FLAMEGPU/FLAMEGPU2/blob/b5173e78765d03be8d1b393f8a037499b064b676/include/flamegpu/model/AgentDescription.h#L227) to be passed as a `std::array`.

I don't think [aggregate init](https://en.cppreference.com/w/cpp/language/aggregate_initialization) permits what you want to do, so you would need to do something like

```c++

std::array<int, 100000> defaultArr;
defaultArr.fill(-1);  // https://en.cppreference.com/w/cpp/container/array/fill
pedestrian.newVariable<int, 100000>(FLOW, defaultArr);
